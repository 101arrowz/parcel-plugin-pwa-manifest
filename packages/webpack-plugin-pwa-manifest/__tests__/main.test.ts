import webpack from 'webpack';
import HtmlWebpackPlugin from 'html-webpack-plugin';
import WebpackPluginPWAManifest from '../index';
import { join } from 'path';
import { readdirSync, readFileSync } from 'fs';
jest.setTimeout(30000);
test('Integrated correctly', async done => {
  const base = join(__dirname, 'sample');
  const out = join(__dirname, 'tmp', Date.now().toString(36));
  await new Promise((res, rej) => {
    webpack({
      mode: 'development',
      plugins: [
        new HtmlWebpackPlugin(),
        new WebpackPluginPWAManifest({
          name: 'epic',
          genIconOpts: {
            baseIcon: join(base, 'icon.svg')
          }
        })
      ],
      output: {
        path: out
      },
      entry: join(base, 'index.js')
    }).run(err => {
      if (err) rej(err);
      else res();
    });
  });
  expect(readdirSync(out)).toMatchSnapshot();
  expect(
    readFileSync(join(out, 'browserconfig.xml')).toString()
  ).toMatchSnapshot();
  expect(readFileSync(join(out, 'index.html')).toString()).toMatchSnapshot();
  expect(
    JSON.parse(readFileSync(join(out, 'manifest.webmanifest')).toString())
  ).toMatchSnapshot();
  done();
});
