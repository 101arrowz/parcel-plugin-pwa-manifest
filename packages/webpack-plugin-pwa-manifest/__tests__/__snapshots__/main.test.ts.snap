// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integrated correctly 1`] = `
Array [
  "apple-touch-icon.a9f2329b.png",
  "browserconfig.xml",
  "icon-152x152.5b713d42.png",
  "icon-152x152.645318f8.webp",
  "icon-192x192.870db1ef.png",
  "icon-192x192.b687c5da.webp",
  "icon-384x384.123b9b28.png",
  "icon-384x384.548009be.webp",
  "icon-512x512.06c9b093.webp",
  "icon-512x512.ecb3e851.png",
  "icon-96x96.54527403.png",
  "icon-96x96.c7f54d5d.webp",
  "index.html",
  "main.js",
  "manifest.webmanifest",
  "mstile-150x150.d71a0330.png",
  "mstile-310x150.b40ebd92.png",
  "mstile-310x310.1fc7ce02.png",
  "mstile-70x70.f4544710.png",
]
`;

exports[`Integrated correctly 2`] = `"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?><browserconfig><msapplication><tile><TileColor>white</TileColor><square70x70logo src=\\"/mstile-70x70.f4544710.png\\"/><square150x150logo src=\\"/mstile-150x150.d71a0330.png\\"/><square310x310logo src=\\"/mstile-310x310.1fc7ce02.png\\"/><wide310x150logo src=\\"/mstile-310x150.b40ebd92.png\\"/></tile></msapplication></browserconfig>"`;

exports[`Integrated correctly 3`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"utf-8\\">
    <title>Webpack App</title>
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\"><meta name=\\"msapplication-config\\" content=\\"/browserconfig.xml\\"><meta name=\\"theme-color\\" content=\\"white\\"><link rel=\\"apple-touch-icon\\" sizes=\\"180x180\\" href=\\"/apple-touch-icon.a9f2329b.png\\"><link rel=\\"manifest\\" href=\\"/manifest.webmanifest\\"></head>
  <body>
  <script src=\\"main.js\\"></script></body>
</html>"
`;

exports[`Integrated correctly 4`] = `
Object {
  "background_color": "white",
  "display": "standalone",
  "icons": Array [
    Object {
      "sizes": "96x96",
      "src": "/icon-96x96.c7f54d5d.webp",
      "type": "image/webp",
    },
    Object {
      "sizes": "96x96",
      "src": "/icon-96x96.54527403.png",
      "type": "image/png",
    },
    Object {
      "sizes": "152x152",
      "src": "/icon-152x152.645318f8.webp",
      "type": "image/webp",
    },
    Object {
      "sizes": "152x152",
      "src": "/icon-152x152.5b713d42.png",
      "type": "image/png",
    },
    Object {
      "sizes": "192x192",
      "src": "/icon-192x192.b687c5da.webp",
      "type": "image/webp",
    },
    Object {
      "sizes": "192x192",
      "src": "/icon-192x192.870db1ef.png",
      "type": "image/png",
    },
    Object {
      "sizes": "384x384",
      "src": "/icon-384x384.548009be.webp",
      "type": "image/webp",
    },
    Object {
      "sizes": "384x384",
      "src": "/icon-384x384.123b9b28.png",
      "type": "image/png",
    },
    Object {
      "sizes": "512x512",
      "src": "/icon-512x512.06c9b093.webp",
      "type": "image/webp",
    },
    Object {
      "sizes": "512x512",
      "src": "/icon-512x512.ecb3e851.png",
      "type": "image/png",
    },
  ],
  "name": "epic",
  "scope": "/",
  "short_name": "epic",
  "start_url": "/",
  "theme_color": "white",
}
`;
